src/tickit/core/components/component.py

```diff
@@ -71,25 +75,38 @@ def __call__(self) -> Component:
 class BaseComponent(Component):
     """A base class for compnents, implementing state interface related methods."""
 
-    state_consumer: StateConsumer[Input]
-    state_producer: StateProducer[Union[Interrupt, Output]]
+    state_consumer: StateConsumer[Union[Input, StopComponent]]
+    state_producer: StateProducer[Union[Interrupt, Output, ComponentException]]
 
-    async def handle_input(self, input: Input):
-        """Calls on_tick when an input is recieved.
+    async def handle_input(self, message: Union[Input, StopComponent]):
+        """Call on_tick when an input is recieved.
 
         Args:
             input (Input): An immutable data container for Component inputs.
         """
-        LOGGER.debug("{} got {}".format(self.name, input))
-        await self.on_tick(input.time, input.changes)
+        if isinstance(message, Input):
+            LOGGER.debug("{} got {}".format(self.name, message))
+            try:
+                await asyncio.gather(
+                    self.on_tick(message.time, message.changes), return_exceptions=False
+                )
+            except Exception as err:
+                LOGGER.error("Exception occured in {} component".format(self.name))
+                LOGGER.error(traceback.format_exc())
```

Comment:
Best to have these as a single call as there is a non-zero possibility they could end up non-adjacent if two separate calls in async code. Also f-strings generally preferred over `.format`